Don't use new Buffer()

RyanZim opened the pull request:
Please review very carefully, as this lib does not have functioning tests.

RyanZim committed:
Don't use new Buffer()

mvayngrib commented on lib/util/buffer.js:
```diff
@@ -109,7 +109,7 @@ module.exports = {
     bytes.push((integer >> 8) & 0xff);
     bytes.push(integer & 0xff);
-    return new Buffer(bytes);
+    return Buffer.alloc(bytes);
```
```suggestion
    return Buffer.from(bytes);
```

RyanZim replied:
This is why we do this review thingy.

RyanZim committed:
alloc > from

Co-authored-by: Mark Vayngrib <mvayngrib@gmail.com>

mvayngrib reviewed (approved):
utACK

kklash reviewed (approved):
utACK

roccomuso reviewed (approved):
tACK on desktop.
- got same address
- send/receive tx

ChALkeR reviewed (dismissed):
This is not complete.

```js
lib/transaction/input/input.js:    prevTxId = new buffer.Buffer(params.prevTxId, 'hex');
lib/transaction/input/input.js:    this._scriptBuffer = new buffer.Buffer(script, 'hex');
lib/transaction/input/input.js:    this._scriptBuffer = new buffer.Buffer(script);
lib/transaction/transaction.js:  this.fromBuffer(new buffer.Buffer(string, 'hex'));
lib/transaction/sighash.js:        atoms: BN.fromBuffer(new buffer.Buffer(BITS_64_ON, 'hex')),
lib/transaction/output.js:        script = new buffer.Buffer(args.script, 'hex');
lib/transaction/output.js:    obj.script = new buffer.Buffer([]);
lib/util/buffer.js:    var result = new buffer.Buffer(bytes);
lib/util/buffer.js:    return new buffer.Buffer([integer & 0xff]);
lib/util/buffer.js:    var ret = new buffer.Buffer(param.length);
lib/util/buffer.js:    return new buffer.Buffer(string, 'hex');
lib/encoding/base58check.js:    data = new buffer.Buffer(Base58.decode(data));
lib/encoding/base58check.js:    checksum = new buffer.Buffer(Base58.decode(checksum));
lib/hdprivatekey.js:    data = BufferUtil.concat([new buffer.Buffer([0]), this.privateKey.toBuffer(), indexBuffer]);
lib/hdprivatekey.js:  var hash = Hash.sha512hmac(hexa, new buffer.Buffer('Bitcoin seed'));
lib/script/script.js:  return new Script(new buffer.Buffer(str, 'hex'));
lib/script/script.js:    return new Script(new buffer.Buffer(str, 'hex'));
```

In all those files:
```js
lib/transaction/input/input.js:var buffer = require('buffer');
lib/transaction/transaction.js:var buffer = require('buffer');
lib/transaction/sighash.js:var buffer = require('buffer');
lib/transaction/output.js:var buffer = require('buffer');
lib/util/buffer.js:var buffer = require('buffer');
lib/encoding/base58.js:var buffer = require('buffer');
lib/encoding/base58check.js:var buffer = require('buffer');
lib/hdprivatekey.js:var buffer = require('buffer');
lib/script/script.js:var buffer = require('buffer');
```

ChALkeR committed:
Fix remaining Buffer() usage

ChALkeR committed:
No reason to use require('buffer')

It was used in same files as Buffer global.

Also, in some functions 'buffer' var was shadowed.

ChALkeR committed:
Use a simpler empty buffer creation method

RyanZim reviewed (commented):
utACK

kklash reviewed (approved):
utACK

roccomuso commented:
tACK

- [x] send/receive

mvayngrib reviewed (approved):
utACK

sonaye commented:
What's the motivation behind these changes, did something break? I'd rather we spend some time on https://github.com/bitcoinjs/bitcoinjs-lib/issues/513 instead

RyanZim commented:
@sonaye This is part of a larger project to remove all usage of `new Buffer` in Exodus desktop. I'm with you, long-term, I'd prefer that solution as well (hate dealing with this lib).

RyanZim commented:
Rebased to fix merge conflicts; @ChALkeR got a few tests running in https://github.com/ExodusMovement/dcrcore-lib/pull/8; they pass.
